### PR TITLE
Adds set text semantics action to render editable

### LINKED ISSUE #77271: Entering text does not work in android voice access
## Steps to Reproduce
```dart
import 'package:flutter/material.dart';
void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Expansion Tile Splash'),
          centerTitle: true,
        ),
        body: TextField(),
      ),
    ),
  );
}
```

1. In android download and turn on voice access
2. In the app, speak "start editing"
3. speak "type how are you"

expect: "how are you" is typed into the text field
actual: nothing happened

### PR DESCRIPTION
**Pull request by chunhtai**
Adds set text semantics action, this is required to support typing in voice access

This requires engine pr: https://github.com/flutter/engine/pull/24734

fixes https://github.com/flutter/flutter/issues/77271

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -3173,6 +3177,25 @@ class SemanticsConfiguration {
...
+  ///
+  /// This handler is invoked when the user either wants to replace the current
+  /// text in the text field with a new text.
+  ///
+  /// Voice access users can trigger this handler by speaking "type <text>" to
+  /// their Android devices.
+  SetTextHandler? get onSetText => _onSetText;
+  SetTextHandler? _onSetText;
+  set onSetText(SetTextHandler? value) {
+    assert(value != null);
```
Comment (justinmc) [inline packages/flutter/lib/src/semantics/semantics.dart:3190]: Nit: Could you remove this assert and make the `value` parameter's type non-nullable?

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -2971,6 +2989,73 @@ void main() {
...
+    expect(
+      semantics,
+      hasSemantics(
+        TestSemantics.root(
+          children: <TestSemantics>[
+            TestSemantics.rootChild(
+              id: 1,
+              children: <TestSemantics>[
+                TestSemantics(
+                    id: 2,
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:3018]: This line is indented too far.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 3]
```diff
@@ -2971,6 +2989,73 @@ void main() {
...
+                            ],
+                            value: '',
+                            textSelection: TextSelection.collapsed(
+                                offset: controller.text.length),
+                            textDirection: TextDirection.ltr,
+                          ),
+                        ],
+                      ),
+                    ]
+                )
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:3042]: Weird indentation here, and commas are missing.

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/semantics/semantics.dart:3190]: The setter and getter need to match, dart won't let you make a non-nullable with nullable getter.

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 4]
```diff
@@ -3173,6 +3177,25 @@ class SemanticsConfiguration {
...
+    _addAction(SemanticsAction.setText, (dynamic args) {
```
[the remark this answers is not in the document]
Comment (chunhtai) [reply packages/flutter/lib/src/semantics/semantics.dart:3191]: Think a little bit more, I think not allowing null value makes more sense, it does not make sense for developer to assign action to semantics configuration in describeSemanticsConfigruation and later on remove it in the same method.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by chunhtai** [comment]
google test fail because it is running against old version of engine.

**Pull request merged**